pkgs/tools/misc/tlp/default.nix

```diff
@@ -1,81 +1,126 @@
-{ stdenv, lib, fetchFromGitHub, perl, makeWrapper, file, systemd, iw, rfkill
-, hdparm, ethtool, inetutils , kmod, pciutils, smartmontools
-, x86_energy_perf_policy, gawk, gnugrep, coreutils, utillinux
-, checkbashisms, shellcheck
-, enableRDW ? false, networkmanager
-}:
-
-let
-  paths = lib.makeBinPath
-          ([ iw rfkill hdparm ethtool inetutils systemd kmod pciutils smartmontools
-             x86_energy_perf_policy gawk gnugrep coreutils utillinux
-           ]
-           ++ lib.optional enableRDW networkmanager
-          );
-
-in stdenv.mkDerivation rec {
+{ stdenv
+, lib
+, checkbashisms
+, coreutils
+, ethtool
+, fetchFromGitHub
+, gawk
+, gnugrep
+, gnused
+, hdparm
+, iw
+, kmod
+, makeWrapper
+, pciutils
+, perl
+, shellcheck
+, smartmontools
+, systemd
+, utillinux
+, x86_energy_perf_policy
+  # RDW only works with NetworkManager, and thus is optional with default off
+, enableRDW ? false
+, networkmanager
+}: stdenv.mkDerivation rec {
   pname = "tlp";
-  version = "1.2.2";
+  version = "1.3.1";
 
   src = fetchFromGitHub {
     owner = "linrunner";
     repo = "TLP";
     rev = version;
-    sha256 = "0vm31ca6kdak9xzwskz7a8hvdp67drfh2zcdwlz3260r8r2ypgg1";
+    sha256 = "14fcnaz9pw534v4d8dddqq4wcvpf1kghr8zlrk62r5lrl46sp1p5";
   };
 
-  outRef = placeholder "out";
+  # XXX: See patch files for relevant explanations.
+  patches = [ ./patches/fix-makefile-sed.patch ./patches/tlp-sleep-service.patch ];
 
+  buildInputs = [ perl ];
+  nativeBuildInputs = [ makeWrapper gnused ];
+
+  # XXX: While [1] states that DESTDIR should not be used, and that the correct
+  # variable to set is, in fact, PREFIX, tlp thinks otherwise. The Makefile for
+  # tlp concerns itself only with DESTDIR [2] (possibly incorrectly) and so we set
+  # that as opposed to PREFIX, despite what [1] says.
+  #
+  # [1]: https://github.com/NixOS/nixpkgs/issues/65718
+  # [2]: https://github.com/linrunner/TLP/blob/ab788abf4936dfb44fbb408afc34af834230a64d/Makefile#L4-L46
   makeFlags = [
-    "DESTDIR=${outRef}"
-    "TLP_SBIN=${outRef}/bin"
-    "TLP_BIN=${outRef}/bin"
-    "TLP_TLIB=${outRef}/share/tlp"
-    "TLP_FLIB=${outRef}/share/tlp/func.d"
-    "TLP_ULIB=${outRef}/lib/udev"
-    "TLP_NMDSP=${outRef}/etc/NetworkManager/dispatcher.d"
-    "TLP_SHCPL=${outRef}/share/bash-completion/completions"
-    "TLP_MAN=${outRef}/share/man"
-    "TLP_META=${outRef}/share/metainfo"
+    "DESTDIR=${placeholder "out"}"
```

Comment:
I think the upstream uses `DESTDIR` correctly (as demonstrated by `fix-makefile-sed.patch`) and the issue is that it does not support [GNU installation directories](http://www.gnu.org/prep/standards/html_node/Directory-Variables.html). Keeping this in mind, the correct solution would be  using `${placeholder "out"}` in place of `${outRef}` and asking upstream to make the path overridability more granular as per GNU guidelines.